### DISCUSSION TITLE
Want to say something about tradingview lightweight-charts

### DISCUSSION THREAD

**Discussion by GitCatacao** [tradingview/lightweight-charts#1792] [General]
I have seen and I used many libraries so far for financial charts, some good, some not so good.

The way you present your "lightweight-charts" are very tempting, but in my opinion something is not as it should be.

<hr>

The fact that the volumes are treated separately creates an epic mess.

When you need to call them in the tooltip via `param.seriesData.get(candlestickSeries);` they no longer belong to the series of type "line" or "candlestick" since the data is excluded even if passed in this way

```

var data = [{ open: 10, high: 10.63, low: 9.49, close: 9.55, volume: 1000, time: 1642427876 }, { open: 9.55, high: 10.30, low: 9.42, close: 9.94, volume: 1000, time: 1642514276 }, { open: 9.94, high: 10.17, low: 9.92, close: 9.78, volume: 1000, time: 1642600676 }, { open: 9.78, high: 10.59, low: 9.18, close: 9.51, volume: 1000, time: 1642687076 }, { open: 9.51, high: 10.46, low: 9.10, close: 10.17, volume: 1000, time: 1642773476 }, { open: 10.17, high: 10.96, low: 10.16, close: 10.47, volume: 1000, time: 1642859876 }, { open: 10.47, high: 11.39, low: 10.40, close: 10.81, volume: 1000, time: 1642946276 }, { open: 10.81, high: 11.60, low: 10.30, close: 10.75, volume: 1000, time: 1643032676 }, { open: 10.75, high: 11.60, low: 10.49, close: 10.93, volume: 1000, time: 1643119076 }, { open: 10.93, high: 11.53, low: 10.76, close: 10.96, volume: 1000, time: 1643205476 }];

```
and an `undefined` value is returned.

See the reference link:
https://tradingview.github.io/lightweight-charts/tutorials/how_to/tooltips#tracking-tooltip

So it would be necessary to look for the timestamp within the data provided and derive the volume from there, which is complicated and requires additional computing power.

The same goes for moving averages that can be easily generated with the functions provided, but which are not passed within the tooltip. This is probably more difficult to solve.

See the reference link:
https://tradingview.github.io/lightweight-charts/tutorials/demos/moving-average

So in my opinion, either you really integrate the volumes, or you don't integrate them, because treating them in this way is a pity.

The other part, the one you provide to the new companies, seems to me to be done with logic, but it's not the one that interests me.

Having said that, congratulations on your company that is truly megagalactic and good work.

I am abandoning the tradingview, lightweight-charts project because after working on it for a day I was dissatisfied.

In case you fix the above I will be a happy user to return to your product.

If so, please provide a complete example of all this with volumes and moving averages in the tooltip to make the developer's life easier.

Another thing I would like to say is that it is complicated to switch from line to candlestick, because line uses the `value` parameter, while candlestick already has the `close`. So every time a datamapping is necessary, when instead the same data provided to the chart could be used. In finance there is no `value` value. It is an `open`, a `close`, a `low` or a `high`.

Regards.

**Comment by hosein1arti** [comment]
Treading viewer

**Comment by GitCatacao** [comment]
@hosein1arti 

> Treading viewer

I have never read such a useless comment in my life. Kudos to you for your effort.

**Comment by SlicedSilver** [comment]
Thank you for your feedback. We will continue our work to improve the library and documentation site, and have taken your feedback on board.
